File: stack/stack_op.java
package stack;
import java.util.*;

class StackDemo {

    static int[] stack = new int[100];
    static int top = -1;

    static void push(int x) {

        if(top == stack.length - 1) {
            System.out.println("Stack Overflow");
            return;
        }

        top++;
        stack[top] = x;

        System.out.println(x + " pushed");
    }

    static void pop() {

        if(top == -1) {
            System.out.println("Stack Underflow");
            return;
        }

        System.out.println(stack[top] + " popped");
        top--;
    }

    static void peek() {

        if(top == -1) {
            System.out.println("Stack is Empty");
            return;
        }

        System.out.println("Top Element: " + stack[top]);
    }

    static void display() {

        if(top == -1) {
            System.out.println("Stack is Empty");
            return;
        }

        for(int i = top; i >= 0; i--) {
            System.out.print(stack[i] + " ");
        }

        System.out.println();
    }

    public static void main(String[] args) {

        push(10);
        push(20);
        push(30);

        display();

        pop();

        peek();

        display();
    }
}

File: stack/stack_op.cpp
#include <iostream>
using namespace std;

int stack[100];
int top = -1;
int stackSize = 100;

void push(int x) {

    if(top == stackSize - 1) {
        cout << "Stack Overflow\n";
        return;
    }

    top++;
    stack[top] = x;

    cout << x << " pushed\n";
}

void pop() {

    if(top == -1) {
        cout << "Stack Underflow\n";
        return;
    }

    cout << stack[top] << " popped\n";
    top--;
}

void peek() {

    if(top == -1) {
        cout << "Stack is Empty\n";
        return;
    }

    cout << "Top Element: " << stack[top] << endl;
}

void display() {

    if(top == -1) {
        cout << "Stack is Empty\n";
        return;
    }

    for(int i = top; i >= 0; i--) {
        cout << stack[i] << " ";
    }

    cout << endl;
}

int main() {

    push(10);
    push(20);
    push(30);

    display();

    pop();

    peek();

    display();

    return 0;
}